File: src/pkg/quote/dto/quote.ts
import Quote from '@/pkg/quote/domain/quote'

export interface IQuoteDto {
  id: string
  content: string
  author: string
}

const convertQuoteFromDomainToDto = (quote: Quote): IQuoteDto => {
  return {
    id: quote._id.toHexString(),
    content: quote.content,
    author: quote.author,
  }
}

export { convertQuoteFromDomainToDto }

File: pkg/common/dto/quote.go
package dto

import "github.com/namhq1989/tapnchill-server/pkg/common/domain"

type Quote struct {
	ID      string `json:"id"`
	Content string `json:"content"`
	Author  string `json:"author"`
}

func (Quote) FromDomain(quote domain.Quote) Quote {
	return Quote{
		ID:      quote.ID,
		Content: quote.Content,
		Author:  quote.Author,
	}
}
